### ISSUE TITLE
[NCPA 2.0.1/Windows Update] Api endpoint system doesnt work

### ISSUE THREAD

**Issue by ccztux** [NagiosEnterprises/ncpa#295]
This issue: #277 still exists when updating from 2.0.0 to 2.0.1.

This are the log messages after a one api call:
```
2017-01-04 09:59:22,888:ERROR:app:Exception on /api/system [GET]
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\flask\app.py", line 1817, in wsgi_app
  File "C:\Python27\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
  File "C:\Python27\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1461, in dispatch_request
  File "C:\ncpa\agent\listener\server.py", line 154, in token_auth_decoration
  File "C:\ncpa\agent\listener\server.py", line 942, in api
  File "C:\Python27\lib\json\__init__.py", line 244, in dumps
  File "C:\Python27\lib\json\encoder.py", line 207, in encode
  File "C:\Python27\lib\json\encoder.py", line 270, in iterencode
UnicodeDecodeError: 'utf8' codec can't decode byte 0xe4 in position 11: invalid continuation byte
2017-01-04 09:59:22,904:INFO:handler:192.168.0.10 - - [2017-01-04 09:59:22] "GET /api/system HTTP/1.1" 500 4145 0.031000
2017-01-04 09:59:22,920:ERROR:app:Exception on /api/system [GET]
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\flask\app.py", line 1817, in wsgi_app
  File "C:\Python27\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
  File "C:\Python27\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1461, in dispatch_request
  File "C:\ncpa\agent\listener\server.py", line 154, in token_auth_decoration
  File "C:\ncpa\agent\listener\server.py", line 942, in api
  File "C:\Python27\lib\json\__init__.py", line 244, in dumps
  File "C:\Python27\lib\json\encoder.py", line 207, in encode
  File "C:\Python27\lib\json\encoder.py", line 270, in iterencode
UnicodeDecodeError: 'utf8' codec can't decode byte 0xe4 in position 11: invalid continuation byte
2017-01-04 09:59:22,920:INFO:handler:192.168.0.10 - - [2017-01-04 09:59:22] "GET /api/system HTTP/1.1" 500 4145 0.016000
```

**Comment by ccztux** [comment]
Its the same situation as described here: https://github.com/NagiosEnterprises/ncpa/issues/277#issuecomment-267708380

**Comment by jomann09** [comment]
I was trying to avoid having to decode but looks like I might have to grab the system encoding and decode it before re-encoding it in utf-8

**Comment by jomann09** [comment]
Unfortunately it looks like this is really only going to work well if we us Python 3 since it seems like a pretty big mess to try to encode/decode different values on Windows - unless someone has a fix for the current version of NCPA. I've tested this in the dev branch for NCPA 3 and it looks like this will work fine when we switch. Unfortunately I am unsure how long it will be for 3 to come out since I just started working on it.
